hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/logaggregation/AppLogAggregatorImpl.java

```diff
@@ -472,7 +476,9 @@ public void run() {
       // do post clean up of log directories on any other exception
       LOG.error("Error occurred while aggregating the log for the application "
           + appId, e);
-      doAppLogAggregationPostCleanUp();
+      if (enableLocalCleanup) {
+        doAppLogAggregationPostCleanUp();
+      }
```

Comment:
As `enableLocalCleanup` isn't local variable, can we skip cleanup in the `doAppLogAggregationPostCleanUp()`?
```
  private void doAppLogAggregationPostCleanUp() {
    if (!enableLocalCleanup) {
      return;
    }
```
That way we can change only one place.

Reply:
Good idea, Addressed in my next commit.